test/ServerSubscriptionTest.php

```diff
@@ -48,30 +52,30 @@ function tearDown()
     function testBasic() {
 
         $subscriptionData = new Subscription();
-        $subscriptionData->accountId =  $this->account->accountId;
-        $subscriptionData->productName = "Sports";
-        $subscriptionData->productCategory = "BASE";
-        $subscriptionData->billingPeriod = "MONTHLY";
-        $subscriptionData->priceList = "DEFAULT";
-        $subscriptionData->externalKey = $this->externalBundleId;
+        $subscriptionData->setAccountId($this->account->getAccountId());
+        $subscriptionData->setProductName('Sports');
+        $subscriptionData->setProductCategory('BASE');
+        $subscriptionData->setBillingPeriod('MONTHLY');
+        $subscriptionData->setPriceList('DEFAULT');
+        $subscriptionData->setExternalKey($this->externalBundleId);
 
         $subscription = $subscriptionData->create($this->user, $this->reason, $this->comment, $this->tenant->getTenantHeaders());
-        $this->assertEquals($subscription->accountId, $subscriptionData->accountId);
-        $this->assertEquals($subscription->productName, $subscriptionData->productName);
-        $this->assertEquals($subscription->productCategory, $subscriptionData->productCategory);
-        $this->assertEquals($subscription->billingPeriod, $subscriptionData->billingPeriod);
-        $this->assertEquals($subscription->externalKey, $subscriptionData->externalKey);
+        $this->assertEquals($subscription->getAccountId(), $subscriptionData->getAccountId());
+        $this->assertEquals($subscription->getProductName(), $subscriptionData->getProductName());
+        $this->assertEquals($subscription->getProductCategory(), $subscriptionData->getProductCategory());
+        $this->assertEquals($subscription->getBillingPeriod(), $subscriptionData->getBillingPeriod());
+        $this->assertEquals($subscription->getExternalKey(), $subscriptionData->getExternalKey());
 
         # Move by a few days -- still in trial -- and change product
         $this->clock->addDays(3, $this->tenant->getTenantHeaders());
-        $subscription->productName = 'Super';
+        $subscription->setProductName('Super');
         $subscriptionRes = $subscription->changePlan(null, null, null, $this->user, $this->reason, $this->comment, $this->tenant->getTenantHeaders());
-        $this->assertEquals($subscriptionRes->productName, 'Super');
+        $this->assertEquals($subscriptionRes->getProductName(), 'Super');
```

Comment:
From looking at the logs it seems that the `CHANGE` event is being processed before the response is sent. I tried to fetch the the subscription in an additional step and then i also moved the clock forward a day in between changing the subscription and fetching it. Both without success.

These are the entries in the `subscription_events` table: 

```
MariaDB [killbill]> select * from subscription_events\G;
*************************** 1. row ***************************
              record_id: 251
                     id: 9f8dd2ee-069c-4436-bccd-21c524f2b58f
             event_type: API_USER
              user_type: CREATE
         effective_date: 2017-04-03 19:42:10
        subscription_id: 7a7fc234-6930-4ff3-a557-20e1e4bd592c
              plan_name: sports-monthly
             phase_name: sports-monthly-trial
        price_list_name: DEFAULT
billing_cycle_day_local: 0
              is_active: 1
             created_by: phpTester
           created_date: 2017-04-03 19:42:10
             updated_by: phpTester
           updated_date: 2017-04-03 19:42:10
      account_record_id: 340
       tenant_record_id: 462
*************************** 2. row ***************************
              record_id: 252
                     id: 08962d52-aedf-4365-92a9-7d45218bb409
             event_type: PHASE
              user_type: NULL
         effective_date: 2017-05-03 19:42:10
        subscription_id: 7a7fc234-6930-4ff3-a557-20e1e4bd592c
              plan_name: NULL
             phase_name: sports-monthly-evergreen
        price_list_name: NULL
billing_cycle_day_local: 0
              is_active: 0
             created_by: phpTester
           created_date: 2017-04-03 19:42:10
             updated_by: phpTester
           updated_date: 2017-04-06 19:42:14
      account_record_id: 340
       tenant_record_id: 462
*************************** 3. row ***************************
              record_id: 253
                     id: ea5c811a-f87a-48e5-8b64-cfbcd75206ce
             event_type: API_USER
              user_type: CHANGE
         effective_date: 2017-04-06 19:42:14
        subscription_id: 7a7fc234-6930-4ff3-a557-20e1e4bd592c
              plan_name: sports-monthly-20
             phase_name: sports-monthly-20-trial
        price_list_name: DEFAULT
billing_cycle_day_local: 0
              is_active: 1
             created_by: phpTester
           created_date: 2017-04-06 19:42:14
             updated_by: phpTester
           updated_date: 2017-04-06 19:42:14
      account_record_id: 340
       tenant_record_id: 462
*************************** 4. row ***************************
              record_id: 254
                     id: f951428a-a3b5-415c-82db-b13d0cfa3e20
             event_type: PHASE
              user_type: NULL
         effective_date: 2017-05-03 19:42:10
        subscription_id: 7a7fc234-6930-4ff3-a557-20e1e4bd592c
              plan_name: NULL
             phase_name: sports-monthly-20-evergreen
        price_list_name: NULL
billing_cycle_day_local: 0
              is_active: 1
             created_by: phpTester
           created_date: 2017-04-06 19:42:14
             updated_by: phpTester
           updated_date: 2017-04-06 19:42:14
      account_record_id: 340
       tenant_record_id: 462
4 rows in set (0.00 sec)
```

Not sure if it's relevant but we're using Killbill 0.18.2